src/Network/Minio/S3API.hs

```diff
@@ -325,3 +329,26 @@ headObject bucket object = do
 
   maybe (throwM MErrVInvalidObjectInfoResponse) return $
     ObjectInfo <$> Just object <*> modTime <*> etag <*> size
+
+
+handleNoSuchBucket :: ServiceErr -> Minio Bool
+handleNoSuchBucket e | e == NoSuchBucket = return False
+                     | otherwise = throwM e
+
+handleStatus404 :: NC.HttpException -> Minio Bool
+handleStatus404 e@(NC.HttpExceptionRequest _ (NC.StatusCodeException res _)) =
+  if NC.responseStatus res == status404
+  then return False
+  else throwM e
+handleStatus404 e = throwM e
+
+-- | Query the object store if a given bucket exists.
+headBucket :: Bucket -> Minio Bool
+headBucket bucket = (do
+  resp <- executeRequest $ def { riMethod = HT.methodHead
+                               , riBucket = Just bucket
+                               }
+  return $ (NC.responseStatus resp) == HT.ok200) `catches`
+                    [ Handler handleNoSuchBucket
+                    , Handler handleStatus404
+                    ]
```

Comment:
Error handlers can be put in a where clause to indicate that they are intended to be used only by this function.